### PR TITLE
feat: async type resolvers

### PR DESCRIPTION
**Pull request by massada**
# Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

Fixes # (issue)

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes.

## Checklist

-   [ ] My code follows the style guidelines of this project
-   [ ] I have performed a self-review of my own code
-   [ ] I have added tests to cover my changes
-   [ ] I have made corresponding changes to the documentation

### Disclaimer

By sending us your contributions, you are agreeing that your contribution is made subject to the terms of our [Contributor Ownership Statement](https://github.com/Farfetch/.github/blob/master/COS.md)

### CODE REVIEW

**Review by filipeesch — APPROVED** [review]

File: src/KafkaFlow.Serializer.SchemaRegistry.ConfluentAvro/ConsumerConfigurationBuilderExtensions.cs [hunk 1]
```diff
@@ -17,9 +17,10 @@ public static class ConsumerConfigurationBuilderExtensions
         public static IConsumerMiddlewareConfigurationBuilder AddSchemaRegistryAvroSerializer(
             this IConsumerMiddlewareConfigurationBuilder middlewares)
         {
-            return middlewares.AddSerializer(
-                resolver => new ConfluentAvroSerializer(resolver),
-                resolver => new SchemaRegistryTypeResolver(new ConfluentAvroTypeNameResolver(resolver.Resolve<ISchemaRegistryClient>())));
+            return middlewares.Add(
```
Comment (alek-talabat) [inline src/KafkaFlow.Serializer.SchemaRegistry.ConfluentAvro/ConsumerConfigurationBuilderExtensions.cs:20]: Why there are no new extension methods to support async versions on https://github.com/Farfetch/kafkaflow/blob/master/src/KafkaFlow.Serializer/ConsumerMiddlewareConfigurationBuilderExtensions.cs ?

Comment (alek-talabat) [reply src/KafkaFlow.Serializer.SchemaRegistry.ConfluentAvro/ConsumerConfigurationBuilderExtensions.cs:20]: @massada @filipeesch

Comment (massada) [reply src/KafkaFlow.Serializer.SchemaRegistry.ConfluentAvro/ConsumerConfigurationBuilderExtensions.cs:20]: Hi @alek-talabat, there were no changes to the public contracts, what do you mean by a breaking change?

Comment (alek-talabat) [reply src/KafkaFlow.Serializer.SchemaRegistry.ConfluentAvro/ConsumerConfigurationBuilderExtensions.cs:20]: @massada  - As `SchemaRegistryTypeResolver` implements `IAsyncMessageTypeResolver` now, it is not possible to use `AddSerializer` (from [here](https://github.com/Farfetch/kafkaflow/blob/master/src/KafkaFlow.Serializer/ConsumerMiddlewareConfigurationBuilderExtensions.cs ))anymore, as it works only with `IMessageTypeResolver`. So after updating the package which was using it, it does not compile. And it's not that straightforward to find how to migrate to the syntax as provided in here.

Comment (alek-talabat) [reply src/KafkaFlow.Serializer.SchemaRegistry.ConfluentAvro/ConsumerConfigurationBuilderExtensions.cs:20]: Yes, I'm using a custom resolver. I thought of not modifying the existing `AddSerializer` methods, but simply adding new ones which would accept generic parameter of type `IAsyncMessageTypeResolver `. I think that should resolve it

Comment (mhkolk) [reply src/KafkaFlow.Serializer.SchemaRegistry.ConfluentAvro/ConsumerConfigurationBuilderExtensions.cs:20]: This change/PR broke the custom type resolvers feature completely. The overloads are there but are not getting called correctly. See #305

**Pull request merged**